libs/wingc/src/jsify.rs

```diff
@@ -578,6 +571,27 @@ impl<'a> JSifier<'a> {
 					}
 				))
 			}
+			StmtKind::Module { name, statements } => {
+				let mut code = CodeMaker::default();
+				code.open(format!("const {} = (() => {{", name.name));
+				code.add_code(self.jsify_scope_body(statements, ctx));
+
+				let exports = get_public_symbols(statements);
+				code.line(format!(
+					"return {{ {} }};",
+					exports.iter().map(ToString::to_string).join(", ")
+				));
+
+				code.close("})();");
+				code
```

Comment:
I mentioned this irl, but just want to make sure it's commented: I don't think inlining the module is sufficient and probably shouldn't wait for a followup change imo. 
- The modules's code will have access to the parent scope. Because we have type checking I'm not 100% sure this will cause issues, but I think there's a chance for sneaky bugs in the emitted js.
- (more important) This will result in duplicate js when the same module is imported in different places. Not only does this contribute to https://github.com/winglang/wing/issues/3511, but it's also very wasteful. It'll also cause issues if there are any `instanceof` checks of classes from modules.

breaking the module out to a new file via a require() instead makes a lot of sense

Reply:
I also share this intuition but I think we can totally iterate on this going forward.

Reply:
> I think we can totally iterate on this going forward.

I would agree if changing this were difficult, but I think switching to a require is a very simple diff. Just emit a `module.exports =` `jsify_scope_body()` to a new file then emit the `require` to the current file.

Reply:
Right now I'm just hijacking the `module` AST node for the use case of multi-file Wing applications -- I'm planning to address these problems as part of the follow up work (handling import cycles etc.) where we won't have to combine all of the ASTs together. If people run into bugs before that, let's create issues to track them.

Reply:
To be clear, this comment was talking about the emitted JS rather the sharing of the AST. I also think that we need to create multiple ASTs for several reasons, but that's a non-trivial change. Emitting the JS separately should be easy and can be done before any AST-weirdness.